airflow/providers/cncf/kubernetes/executors/kubernetes_executor_utils.py

```diff
@@ -139,7 +139,9 @@ def _run(
     ) -> str | None:
         self.log.info("Event: and now my watch begins starting at resource_version: %s", resource_version)
 
-        kwargs: dict[str, Any] = {"label_selector": f"airflow-worker={scheduler_job_id}"}
+        kwargs: dict[str, Any] = {
+            "label_selector": f"airflow-worker={scheduler_job_id},{POD_EXECUTOR_DONE_KEY}=False",
```

Comment:
Since the label is not there from start, and added at some point, won't it prevent listing pods without the label? I didn't experiment with that but will do cause that was my thinking when I added it